apps/prairielearn/assets/scripts/lib/histogram.ts

```diff
@@ -0,0 +1,125 @@
+import { scaleLinear, axisBottom, axisLeft, select } from 'd3';
+
+export function histogram(
+  selector: Element,
+  data: number[],
+  xgrid: number[],
+  options: {
+    width?: number;
+    height?: number;
+    xmin?: number | 'auto';
+    xmax?: number | 'auto';
+    ymin?: number | 'auto';
+    ymax?: number | 'auto';
+    xlabel?: string;
+    ylabel?: string;
+    xTickLabels?: string[] | 'auto';
+    topMargin?: number;
+    rightMargin?: number;
+    bottomMargin?: number;
+    leftMargin?: number;
+  },
+) {
+  const resolvedOptions = {
+    width: 100,
+    height: 40,
+    xmin: 'auto',
+    xmax: 'auto',
+    ymin: 'auto',
+    ymax: 'auto',
+    xlabel: 'value',
+    ylabel: 'count',
+    xTickLabels: 'auto',
+    topMargin: 10,
+    rightMargin: 20,
+    bottomMargin: 55,
+    leftMargin: 70,
+    ...options,
+  };
+
+  const width = 600 - (resolvedOptions.leftMargin ?? 70) - (resolvedOptions.rightMargin ?? 20);
+  const height = 371 - (resolvedOptions.topMargin ?? 10) - (resolvedOptions.bottomMargin ?? 55);
```

Comment:
Could these hard-coded values be declared as constants at the top of the file?

Reply:
I have moved these hard coded values to the default values for `resolvedOptions.width` and `resolvedOptions.height` and pass those values in here. This allows us to change these if we want to in future uses of this `lib/histogram`.